src/irmin-pack/unix/file_manager.ml

```diff
@@ -797,4 +797,61 @@ struct
     let chunk_start_idx = pl.chunk_start_idx in
     let chunk_num = pl.chunk_num in
     cleanup ~root ~generation ~chunk_start_idx ~chunk_num
+
+  let create_one_commit_store config ~generation ~new_store_root
+      ~latest_gc_target_offset ~suffix_start_offset commit_key =
+    let open Result_syntax in
+    let root = Irmin_pack.Conf.root config in
```

Comment:
To reinforce the idea that the `config` is passed as a way of creating a new store, I think it is slightly confusing to use its `root` value as the source. In my mind, it is more consistent to have something like:

```ocaml
let src_root = t.root in
let dst_root = Irmin_pack.Conf.root config in
...
```
and to drop passing `new_store_root`. 

Or, alternatively, we could drop passing `config`, use only default values for opening various files, and have input parameters like `~src_root` and `~new_store_root | ~dst_root`.

WDYT?

Reply:
I'm passing only config now.